tests/datatype/tests_String.cpp

```diff
@@ -24,11 +24,6 @@ TEST_CASE("Datatype String") {
     auto type_iri = IRI(datatypes::xsd::String::identifier);
     CHECK(type_iri.identifier() == correct_type_iri_cstr);
 
```

Comment:
We need some tests for the native-type storage part. My suggestion would be to make things protected instead of private and inherit from it for the tests to gain access to internals.

Reply:
I went for a different, more indirect approach. Instead of exposing the internals I am now checking Literal construction and value extractions for consistency which is effectively the same thing, because an improperly implemented backend would sometimes return the wrong type of view or the wrong type of value.